tests/func/test_odb.py

```diff
@@ -41,144 +41,144 @@ def test_get(self):
-        self.assertEqual(len(ret), 0)
+def test_cache_load_bad_dir_cache(tmp_dir, dvc):
+    from dvc.data import load
 
-    def test(self):
-        from dvc.data import load
+    dir_hash = "123.dir"
+    fname = os.fspath(dvc.odb.local.hash_to_path(dir_hash))
+    tmp_dir.gen({fname: "<clearly>not,json"})
+    with pytest.raises(ObjectFormatError):
+        load(dvc.odb.local, HashInfo("md5", dir_hash))
 
-        dir_hash = "123.dir"
-        fname = os.fspath(self.dvc.odb.local.hash_to_path(dir_hash))
-        self.create(fname, "<clearly>not,json")
-        with pytest.raises(ObjectFormatError):
-            load(self.dvc.odb.local, HashInfo("md5", dir_hash))
+    dir_hash = "234.dir"
+    fname = os.fspath(dvc.odb.local.hash_to_path(dir_hash))
+    tmp_dir.gen({fname: '{"a": "b"}'})
+    with pytest.raises(ObjectFormatError):
+        load(dvc.odb.local, HashInfo("md5", dir_hash))
 
-        dir_hash = "234.dir"
-        fname = os.fspath(self.dvc.odb.local.hash_to_path(dir_hash))
-        self.create(fname, '{"a": "b"}')
-        with pytest.raises(ObjectFormatError):
-            load(self.dvc.odb.local, HashInfo("md5", dir_hash))
 
+def test_external_cache_dir(tmp_dir, dvc, make_tmp_dir):
+    cache_dir = make_tmp_dir("cache")
 
-class TestExternalCacheDir(TestDvc):
-    def test(self):
-        cache_dir = TestDvc.mkdtemp()
+    ret = main(["config", "cache.dir", cache_dir.fs_path])
+    assert ret == 0
 
-        ret = main(["config", "cache.dir", cache_dir])
-        self.assertEqual(ret, 0)
+    assert not os.path.exists(dvc.odb.local.cache_dir)
 
-        self.assertFalse(os.path.exists(self.dvc.odb.local.cache_dir))
+    tmp_dir.gen({"foo": "foo"})
+    ret = main(["add", "foo"])
+    assert ret == 0
 
-        ret = main(["add", self.FOO])
-        self.assertEqual(ret, 0)
+    tmp_dir.gen(
+        {
+            "data_dir": {
+                "data": "data_dir/data",
+                "data_sub_dir": {"data_sub": "data_dir/data_sub_dir/data_sub"},
+            }
+        }
+    )
+    ret = main(["add", "data_dir"])
+    assert ret == 0
 
-        ret = main(["add", self.DATA_DIR])
-        self.assertEqual(ret, 0)
+    assert not os.path.exists(".dvc/cache")
+    assert len(os.listdir(cache_dir)) != 0
 
-        self.assertFalse(os.path.exists(".dvc/cache"))
-        self.assertNotEqual(len(os.listdir(cache_dir)), 0)
 
-    def test_remote_references(self):
-        ssh_url = "ssh://user@localhost:23"
-        assert main(["remote", "add", "storage", ssh_url]) == 0
-        assert main(["remote", "add", "cache", "remote://storage/tmp"]) == 0
-        assert main(["config", "cache.ssh", "cache"]) == 0
+def test_remote_references(tmp_dir, dvc):
+    ssh_url = "ssh://user@localhost:23"
+    assert main(["remote", "add", "storage", ssh_url]) == 0
+    assert main(["remote", "add", "cache", "remote://storage/tmp"]) == 0
+    assert main(["config", "cache.ssh", "cache"]) == 0
 
-        self.dvc.__init__()
+    dvc.__init__()
```

Comment:
@pmrowla - I've removed `dvc.__init__` call and `main` calls in this test. But I had to also add `dvc.odb = ODBManager(dvc)` because `dvc.config` updates are not applied to `dvc.odb.config`. If there is a better way to update `odb`, please tell me.